crispy_forms/utils.py

```diff
@@ -102,16 +90,19 @@ def render_field(
                 for index, (widget, attr) in enumerate(zip(widgets, list_attrs)):
                     if hasattr(field_instance.widget, 'widgets'):
                         if 'type' in attr and attr['type'] == "hidden":
-                            set_hidden(field_instance.widget.widgets[index])
-                            field_instance.widget.widgets[index] = field_instance.hidden_widget()
+                            field_instance.widget.widgets[index] = field_instance.hidden_widget(attr)
 
-                        field_instance.widget.widgets[index].attrs.update(attr)
+                        else:
+                            # TODO: can we just injext when instantiating?
+                            # where does this happen (here for hidden, where
+                            # for regular?)
+                            field_instance.widget.widgets[index].attrs.update(attr)
```

Comment:
django 1.11 pops `type` from attrs in `__init__`. Just overriding the attrs later ended up with `type=hidden` declared twice